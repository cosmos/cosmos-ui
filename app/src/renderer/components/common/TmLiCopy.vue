<template>
  <div class="tm-li-copy">
    <div class="value">
      {{ value }}
    </div>
    <menu> <btn-copy :value="value" /> </menu>
  </div>
</template>

<script>
import BtnCopy from "common/TmBtnCopy"
export default {
  name: `tm-li-copy`,
  components: { BtnCopy },
  props: {
    value: {
      type: String,
      required: true
    }
  }
}
</script>

<style>
.tm-li-copy {
  height: 3rem;
  display: flex;
  align-items: center;
  border-bottom: px solid var(--bc-dim);
}

.tm-li-copy:only-child {
  border-bottom: none;
}

.tm-li-copy .value {
  flex: 1;
  padding: 0 1rem;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.tm-li-copy menu {
  display: flex;
  align-items: center;
  padding: 0 0.75em;
}
</style>
