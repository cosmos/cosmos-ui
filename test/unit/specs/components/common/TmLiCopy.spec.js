import setup from "../../../helpers/vuex-setup"
import TmLiCopy from "common/TmLiCopy"

describe(`TmLiCopy`, () => {
  let wrapper
  const { mount } = setup()

  beforeEach(() => {
    const instance = mount(TmLiCopy, {
      propsData: {
        value: `hallo`,
        receive: false
      }
    })
    wrapper = instance.wrapper
  })

  it(`has the expected html structure`, () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  it(`should show the provided value`, () => {
    wrapper.setProps({ value: `HALLO` })
    expect(wrapper.html()).toContain(`HALLO`)
  })

  it(`should show the receive button`, () => {
    wrapper.setProps({ value: `HALLO`, receive: true })
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
