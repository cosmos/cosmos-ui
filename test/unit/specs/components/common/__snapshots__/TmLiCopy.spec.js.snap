// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmLiCopy has the expected html structure 1`] = `
<div
  class="tm-li-copy"
>
  <div
    class="value"
  >
    
    hallo
  
  </div>
   
  <menu>
    <button
      class="tm-btn btn-copy"
    >
      <span
        class="tm-btn__container"
      >
        <i
          aria-hidden="true"
          class="tm-btn__icon material-icons"
        >
          
      content_copy
    
        </i>
        <span
          class="tm-btn__value"
        >
          Copy
        </span>
      </span>
    </button>
  </menu>
</div>
`;

exports[`TmLiCopy should show the receive button 1`] = `
<div
  class="tm-li-copy"
>
  <div
    class="value"
  >
    
    HALLO
  
  </div>
   
  <menu>
    <button
      class="tm-btn btn-copy"
    >
      <span
        class="tm-btn__container"
      >
        <i
          aria-hidden="true"
          class="tm-btn__icon material-icons"
        >
          
      content_copy
    
        </i>
        <span
          class="tm-btn__value"
        >
          Copy
        </span>
      </span>
    </button>
  </menu>
</div>
`;
